contrib/signet/getcoins.py

```diff
@@ -5,21 +5,36 @@
 
 import argparse
 import subprocess
-import requests
 import sys
+import requests
+
+DEFAULT_GLOBAL_FAUCET = 'https://signetfaucet.com/claim'
+GLOBAL_FIRST_BLOCK_HASH = '00000086d6b2636cb2a392d45edc4ec544a10024d30141c9adf4bfd9de533b53'
 
 parser = argparse.ArgumentParser(description='Script to get coins from a faucet.', epilog='You may need to start with double-dash (--) when providing bitcoin-cli arguments.')
 parser.add_argument('-c', '--cmd', dest='cmd', default='bitcoin-cli', help='bitcoin-cli command to use')
-parser.add_argument('-f', '--faucet', dest='faucet', default='https://signetfaucet.com/claim', help='URL of the faucet')
+parser.add_argument('-f', '--faucet', dest='faucet', default=DEFAULT_GLOBAL_FAUCET, help='URL of the faucet')
 parser.add_argument('-a', '--addr', dest='addr', default='', help='Bitcoin address to which the faucet should send')
 parser.add_argument('-p', '--password', dest='password', default='', help='Faucet password, if any')
 parser.add_argument('bitcoin_cli_args', nargs='*', help='Arguments to pass on to bitcoin-cli (default: -signet)')
 
 args = parser.parse_args()
 
+if args.bitcoin_cli_args == []:
+    args.bitcoin_cli_args = ['-signet']
+
+if args.faucet.lower() == DEFAULT_GLOBAL_FAUCET:
+    # Get the hash of the block at height 1 of the currently active signet chain
+    try:
+        curr_signet_hash = subprocess.check_output([args.cmd] + args.bitcoin_cli_args + ['getblockhash', '1']).strip().decode()
+    except FileNotFoundError:
+        print('The binary', args.cmd, 'could not be found.')
+        exit()
```

Comment:
nit: When using a SigNet with a chain length of 1 (chain = only the Genesis block), this will fail with a `Block height out of range` error message. This might be confusing for someone unfamiliar with the `getcoins.py` code. If you happen to touch this again, you could add catch-all except handling here mentioning that checking the hash of the first block to determine the SigNet chain failed. This should also print the RPC response with `Block height out of range`.